api/submit_result.go

```diff
@@ -32,12 +36,32 @@ func (s *resultSubmitter) Submit(executionID string, outputKey string, outputDat
 	if err != nil {
 		return err
 	}
-	if err := exec.Complete(outputKey, outputData); err != nil {
-		return err
+
+	// parse output data into a map.
+	var outputDataMap map[string]interface{}
+	switch d := outputData.(type) {
```

Comment:
Change outputData to slice of []bytes as it looks like a dynamic programming language. Here we should expect only one type as discussed.

Reply:
done